#2889 added quick-start tutorial to documentation

kefeimo opened the pull request:
# Description

Please include a summary of the change and which issue is fixed
- Added `quick-start.rst`
- Modified `index.rst` to reflect the change in content structure.

Please also include relevant motivation and context.
- Provide an easy-to-follow tutorial for first-time users.

List any dependencies that are required for this change.
- None

Fixes #2889 (Documentation update) 

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [x] Content review
- [x] Rendering test

1. `index.rst` rendering, (readthedocs entry): https://volttron-km-readthedocs.readthedocs.io/en/latest/
![Picture1](https://user-images.githubusercontent.com/28743873/158239888-89b4e79d-11a0-4f8d-822f-2e5fe7a32eb9.png)

2. `quick-start.rst` rendering, (core content):
https://volttron-km-readthedocs.readthedocs.io/en/latest/developing-volttron/quick-start.html
![image](https://user-images.githubusercontent.com/28743873/158240853-df128e24-cd3b-4def-bcba-531b7269d46c.png)


# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules

craig8 commented on docs/source/developing-volttron/quick-start.rst:
```diff
@@ -0,0 +1,331 @@
+
+You may notice the command prompt has changed and there is the virtual environment name as prefix. e.g., :code:`(volttron) user@host:~/volttron $`. The prefix environment name indicates the virtual environment is activated.
+
+Alternatively, you can use the following command to verify if the virtual environment is up.
```
This assumes that they don't have a different virtual environment setup such as if you have a conda environment then you already have a virtual env setup.

kefeimo replied:
Resolve: Added the following note-block
![image](https://user-images.githubusercontent.com/28743873/158310966-79d8c24e-df38-48d3-8812-eef5e1081ec2.png)

craig8 commented on docs/source/developing-volttron/quick-start.rst:
```diff
@@ -0,0 +1,331 @@
+       # remember to activate the virtual environment again
+       # to follow the rest of the steps
+
+       # deactivate volttron
```
$ rather than # you can't run as root user.

kefeimo replied:
(Originally, I used `#` to indicate the comment-out, but I can see the representation is confusing.)
Resolve with the following modification:
```
Use `deactivate` command to deactivate the virtual environment, i.e., :code:`$ deactivate volttron`. Note: if you run this command, remember to re-activate the virtual environment to follow the rest of the steps.         
```   
![image](https://user-images.githubusercontent.com/28743873/158312787-67791922-4bc2-4b57-820c-da694720221f.png)

craig8 reviewed (changes requested):
I like what you have done here.

craig8 commented:
We need to update the README for the volttron repository to move the rmq to a link and perhaps link to the quick start tutorial but still include the basic things in the readme.

kefeimo commented:
Changes:
1. Moved quick-start a new folder called "Tutorials"
![image](https://user-images.githubusercontent.com/28743873/158313143-1803b009-334e-42a5-9071-9e53c280f907.png)

2. Changed narratives to avoid confusion. (revision as follows.)
![image](https://user-images.githubusercontent.com/28743873/158313281-5742cbce-ab7b-425c-8140-fc90fad8df0d.png)

3. Added note-block to make explicit assumption that the tutorial is under `virtualenv` (instead of, say, `conda`).
![image](https://user-images.githubusercontent.com/28743873/158313461-19d4a385-3a37-4d54-a77a-8747622db45c.png)

awoz commented:
Consider adding a short Deployment section or Next Steps link to "Setting up VOLTTRON as a System Service". 
I found that after learning the Quick Start operations, a developer would want to auto-run the VOLTTRON platform and demo agents.
